src/library/compiler/inliner.cpp

```diff
@@ -129,11 +139,62 @@ class inline_simple_definitions_fn : public compiler_step_visitor {
         return ctx().reduce_projection(e);
     }
 
-    virtual expr visit_app(expr const & e) override {
-        expr const & fn = get_app_fn(e);
+    optional<expr> apply_overrides_rec(expr const & e) {
+        if (is_constant(e)) {
+            name n = const_name(e);
+            if (auto override_name = get_vm_override_name(m_env, n, m_enable_overrides)) {
+                n = override_name.value();
+                while (auto override_name = get_vm_override_name(m_env, n, m_enable_overrides)) {
+                    n = override_name.value();
+                }
+                return optional<expr>(mk_constant(n, const_levels(e)));
+            }
+        } else if (is_app(e)) {
+            if (auto fn = apply_overrides_rec(app_fn(e))) {
+                return optional<expr>(mk_app(*fn, app_arg(e)));
+            }
+        }
+        return optional<expr>();
+    }
+
+    expr apply_overrides(expr const & e) {
+        if (m_enable_overrides) {
+            if (auto e2 = apply_overrides_rec(e)) {
+                return *e2;
+            } else {
+                return e;
+            }
+        } else {
+            return e;
+        }
+    }
+
+    virtual expr visit_constant(expr const & e) override {
+        if (m_enable_overrides) {
+            name n = const_name(e);
+            if (auto override_name = get_vm_override_name(m_env, n, m_enable_overrides)) {
```

Comment:
It would be a good idea to factor this out into a separate function, maybe like this:
```c++
 optional<name> get_vm_override_name_trans(environment const & env, name const & decl_name);
```

Reply:
Good point. I have already this kind of transitivity in `get_vm_decl` so I'll just delegate it to that group of functions